maven-artifact/src/main/java/org/apache/maven/artifact/versioning/ComparableVersion.java

```diff
@@ -211,6 +235,30 @@ public int compareTo( Item item )
             }
         }
 
+        @Override
+        public boolean equals( Object o )
+        {
+            if ( this == o )
+            {
+                return true;
+            }
+            if ( o == null || getClass() != o.getClass() )
+            {
+                return false;
+            }
+
+            LongItem longItem = (LongItem) o;
+
+            return value == longItem.value;
+
+        }
+
+        @Override
+        public int hashCode()
```

Comment:
Why not use `Objects#hashCode()` on this?

Reply:
What for? It's a single object, not a list.